File: services/core-api/internal/features/platformstatus/service_test.go
package platformstatus

import (
	"context"
	"errors"
	"testing"
	"time"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

// fakeRepo is a hand-rolled fake of the StatusReader seam (research B9: no gomock).
type fakeRepo struct {
	status PlatformStatus
	err    error
}

func (f *fakeRepo) Status(_ context.Context) (PlatformStatus, error) {
	return f.status, f.err
}

func TestServiceGet(t *testing.T) {
	dbTime := time.Date(2026, 7, 5, 12, 0, 0, 0, time.UTC)

	tests := []struct {
		name    string
		repo    *fakeRepo
		wantErr bool
		want    PlatformStatus
	}{
		{
			name: "stamps the environment onto the repository read",
			repo: &fakeRepo{status: PlatformStatus{
				DatabaseName: "effy", DatabaseTime: dbTime,
				MigrationVersion: 20260705095817, MigrationsApplied: 1,
			}},
			want: PlatformStatus{
				Environment: "dev", DatabaseName: "effy", DatabaseTime: dbTime,
				MigrationVersion: 20260705095817, MigrationsApplied: 1,
			},
		},
		{
			name:    "propagates repository errors untouched",
			repo:    &fakeRepo{err: errors.New("relation goose_db_version does not exist")},
			wantErr: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			svc := NewService(tt.repo, "dev")
			got, err := svc.Get(context.Background())

			if tt.wantErr {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: apis/edge-api/store/src/status/service.test.ts
import { describe, expect, it } from "vitest";

import type { StatusRepository } from "./repository";
import { createPlatformStatusService } from "./service";

const dbTime = new Date("2026-07-05T12:00:00Z");

function fakeRepo(overrides?: Partial<Awaited<ReturnType<StatusRepository["status"]>>>): StatusRepository {
  return {
    status: () =>
      Promise.resolve({
        databaseName: "effy",
        databaseTime: dbTime,
        migrationVersion: 20260705095817,
        migrationsApplied: 1,
        ...overrides,
      }),
  };
}

describe("platform status service", () => {
  it("stamps the environment onto the repository read", async () => {
    const svc = createPlatformStatusService(fakeRepo(), "dev");
    await expect(svc.getStatus()).resolves.toEqual({
      environment: "dev",
      databaseName: "effy",
      databaseTime: dbTime,
      migrationVersion: 20260705095817,
      migrationsApplied: 1,
    });
  });

  it("propagates repository errors untouched (handlers own the mapping)", async () => {
    const svc = createPlatformStatusService(
      { status: () => Promise.reject(new Error('relation "goose_db_version" does not exist')) },
      "dev",
    );
    await expect(svc.getStatus()).rejects.toThrow("goose_db_version");
  });
});
